extensions/skyportal/skyportal/handlers/api/alert.py

```diff
@@ -226,26 +223,134 @@ async def get(self, objectId: str = None):
+            # either all or none should be provided
+            if ra is None:
+                return self.error("Missing required parameter: ra")
+            if dec is None:
+                return self.error("Missing required parameter: dec")
+            if radius is None:
+                return self.error("Missing required parameter: radius")
+            if radius_units is None:
+                return self.error("Missing required parameter: radius_units")
+        if all(position_tuple):
+            # complete positional arguments? run "near" query
+            if radius_units not in ["deg", "arcmin", "arcsec"]:
+                return self.error(
+                    "Invalid radius_units value. Must be one of either "
+                    "'deg', 'arcmin', or 'arcsec'."
+                )
+            try:
+                ra = float(ra)
+                dec = float(dec)
+                radius = float(radius)
+            except ValueError:
+                return self.error("Invalid (non-float) value provided.")
+            if (
+                (radius_units == "deg" and radius > 1)
+                or (radius_units == "arcmin" and radius > 60)
+                or (radius_units == "arcsec" and radius > 3600)
+            ):
+                return self.error("Radius must be <= 1.0 deg")
+
+            query = {
+                "query_type": "near",
+                "query": {
+                    "max_distance": radius,
+                    "distance_units": radius_units,
+                    "radec": {"query_coords": [ra, dec]},
+                    "catalogs": {
+                        "ZTF_alerts": {
+                            "filter": {
+                                "candidate.programid": {"$in": program_id_selector},
+                            },
+                            "projection": default_projection
+                            if not include_all_fields
+                            else {
+                                "_id": 0,
+                                "cutoutScience": 0,
+                                "cutoutTemplate": 0,
+                                "cutoutDifference": 0,
+                            },
+                        }
+                    },
+                },
+                "kwargs": {
+                    "max_time_ms": 10000,
+                    "limit": 10000,
+                },
+            }
+
+            # additional filters?
+            if object_id is not None:
+                query["query"]["catalogs"]["ZTF_alerts"]["filter"][
+                    "objectId"
+                ] = object_id
+
+            response = kowalski.query(query=query)
+
+            if response.get("status", "error") == "success":
+                alert_data = response.get("data")
+                return self.success(data=alert_data["ZTF_alerts"]["query_coords"])
+
+            return self.error(response("message"))
+
+        # otherwise, run a find query with the specified filter
```

Comment:
Correct me if I'm wrong here, but we won't ever reach this block right? Either `object_id is not None` and it returns in the aggregate query case, or it goes into the positional query and either does the cone search or fails due to missing positional parameters right?

Reply:
:+1:

Reply:
No, it checks if `object_id` was not set in path, there are three options:
1) Only positional query was specified - this is the equivalent of @acrellin's `alerts_by_coords`
2) `objectId` was specified in query, but no positional stuff. I want to keep it this way so that in the future, we could add querying by other parameters, or multiple `objectId`'s in one go
3) Both `objectId` and positional stuff is specified. Then both are combined in the query. Again, this will be very  useful in the future when 2) is expanded.

I additionally check whether all required positional info was provided - this is where having something like `odmantic` for query spec validation that we use in K would be nice to have; but is fine as is, I guess.

Reply:
Ah I see - I missed the fact that it was both a path and query param 👍